packages/fulfillment/src/services/fulfillment-module-service.ts

```diff
@@ -189,42 +186,55 @@ export default class FulfillmentModuleService<
     config: FindConfig<ShippingOptionDTO> = {},
     @MedusaContext() sharedContext: Context = {}
   ): Promise<FulfillmentTypes.ShippingOptionDTO[]> {
-    const {
-      filters: normalizedFilters,
-      config: normalizedConfig,
-      context,
-    } = FulfillmentModuleService.normalizeShippingOptionsListParams(
-      filters,
-      config
-    )
+    const { filters: normalizedFilters, config: normalizedConfig } =
+      FulfillmentModuleService.normalizeShippingOptionsListParams(
+        filters,
+        config
+      )
 
-    let shippingOptions = await this.shippingOptionService_.list(
+    const shippingOptions = await this.shippingOptionService_.list(
       normalizedFilters,
       normalizedConfig,
       sharedContext
     )
 
-    // Apply rules context filtering
-    if (context) {
-      shippingOptions = shippingOptions.filter((shippingOption) => {
-        if (!shippingOption.rules?.length) {
-          return true
-        }
-
-        return isContextValid(
-          context,
-          shippingOption.rules.map((r) => r)
-        )
-      })
-    }
-
     return await this.baseRepository_.serialize<
       FulfillmentTypes.ShippingOptionDTO[]
     >(shippingOptions, {
       populate: true,
     })
   }
 
+  @InjectManager("baseRepository_")
+  // @ts-ignore
+  async listAndCountShippingOptions(
+    filters: FilterableShippingOptionProps = {},
+    config: FindConfig<ShippingOptionDTO> = {},
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<[ShippingOptionDTO[], number]> {
+    const { filters: normalizedFilters, config: normalizedConfig } =
```

Comment:
Why do we need the custom implementation here? I know there is a good reason, but what is it? 😄

Reply:
With the separate method `listShippingOptionsForContext` maybe we don't need this?

I would expect `listShippingOptions` to handle filters the same way other list methods do.

Reply:
Ahah there is no good reason, it is just because of the shortcut fulfillment_set_id and so on which means that we need to re map them.
But i ll wait until we settle on the approach and then cleanup.